apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx

```diff
@@ -0,0 +1,168 @@
+import {Quantities} from '../../../../yoroi-wallets/utils'
+import {useStrings} from '../../common/strings'
+
+export const TransactionSummary = ({feesInfo, buyToken, sellToken}) => {
+  const [bottomSheetState, setBottomSheetSate] = React.useState<{isOpen: boolean; title: string; content?: string}>({
+    isOpen: false,
+    title: '',
+    content: '',
+  })
+  const strings = useStrings()
+  const wallet = useSelectedWallet()
+
+  return (
+    <View>
+      <View>
+        <View style={styles.card}>
+          <Text style={styles.cardText}>{strings.total}</Text>
+
+          <View>
+            <Text style={[styles.cardText, styles.cardTextValue]}>{`${Quantities.format(
+              buyToken.quantity,
+              buyToken.decimals ?? 0,
+            )} ${buyToken.name}`}</Text>
+
+            <Spacer height={6} />
+
+            <Text style={styles.cardTextUSD}></Text>
+          </View>
+        </View>
+
+        <Spacer height={24} />
+
+        {feesInfo.map((orderInfo) => {
+          return (
+            <View key={orderInfo.label}>
+              <Spacer height={8} />
+
+              <View style={styles.flexBetween}>
+                <View style={styles.flex}>
+                  <Text style={[styles.text, styles.gray]}>{orderInfo.label}</Text>
+
+                  <Spacer width={8} />
+
+                  <TouchableOpacity
+                    onPress={() => {
+                      setBottomSheetSate({
+                        isOpen: true,
+                        title: orderInfo.label,
+                        content: orderInfo.info,
+                      })
+                    }}
+                  >
+                    <Icon.Info size={24} />
+                  </TouchableOpacity>
+                </View>
+
+                <Text style={styles.text}>{orderInfo.value}</Text>
+              </View>
+            </View>
+          )
+        })}
+
+        <Spacer height={24} />
+
+        <Text style={styles.amountItemLabel}>{strings.swapFrom}</Text>
+
+        <AmountItem wallet={wallet} amount={{tokenId: sellToken.id, quantity: sellToken.quantity}} />
+
+        <Spacer height={16} />
+
+        <Text style={styles.amountItemLabel}>{strings.swapTo}</Text>
+
+        <AmountItem wallet={wallet} amount={{tokenId: buyToken.id, quantity: buyToken.quantity}} />
+      </View>
+
+      <BottomSheetModal
+        isOpen={bottomSheetState.isOpen}
+        title={bottomSheetState.title}
+        content={<Text style={styles.text}>{bottomSheetState.content}</Text>}
+        onClose={() => {
+          setBottomSheetSate({isOpen: false, title: '', content: ''})
+        }}
+      />
+    </View>
+  )
+}
+
+export const useSwapTx = (
+  {wallet, entry}: {wallet: YoroiWallet; entry: YoroiEntry},
+  options?: UseQueryOptions<YoroiUnsignedTx, Error, YoroiUnsignedTx, [string, 'swap-tx']>,
+) => {
+  console.log('SWAP ENTRY entry', entry)
+  const query = useQuery({
```

Comment:
the yoroiUnsignedTx is an object that is available in the state so you can read the data from there - fee for example, the rest should be available in the state